compiler/rustc_lexer/src/unescape.rs

```diff
@@ -263,6 +259,7 @@ fn ascii_check(first_char: char, mode: Mode) -> Result<char, EscapeError> {
 }
 
 fn unescape_char_or_byte(chars: &mut Chars<'_>, mode: Mode) -> Result<char, EscapeError> {
+    assert!(mode == Mode::Char || mode == Mode::Byte);
```

Comment:
Looking at those asserts, I am wondering if they should be `debug_asserts`? We call this from a pub generic function, so I am not sure that compiler would be able to inline this and see that there's no panicking branches.